linux_docker_resources/Dockerfile-RHEL

```diff
@@ -42,7 +42,7 @@ RUN dnf update \
 RUN rosdep init && rosdep update
 
 # Install some stuff via Pip which we don't have available in EPEL
-RUN python3 -m pip install mypy==0.761 pydocstyle
+RUN python3 -m pip install pydocstyle
```

Comment:
I don't see mypy in the system dependencies installed later.
But because the other changes in this branch will install rosdistro-dependent versions of mypy dropping it here seems reasonable.

Reply:
Ah yes, RHEL is an interesting case.  RHEL does not package mypy, either in the main repositories or in EPEL.  So we have to install it from pip (which this patch nicely accomplishes without special cases).

As a side note, we get away with releasing `ament_mypy` into RHEL on https://build.ros2.org because we specifically patch out mypy as a dependency there.  It means that an `dnf install ros-rolling-ament-mypy` on RHEL doesn't result in a fully-functional system, but that was deemed an acceptable tradeoff until the day that mypy gets packaged into RHEL.